ui/app/AppLayouts/Wallet/popups/filterSubMenus/ActivityTypeFilterSubMenu.qml

```diff
@@ -44,43 +34,52 @@ StatusMenu {
     }
 
     ActivityTypeCheckBox {
+        id: sendCheckbox
         title: qsTr("Send")
-        asset.name: "send"
-        checked: typeFilters.includes(ActivityTypeFilterSubMenu.Send)
+        assetSettings.name: "send"
         buttonGroup: typeButtonGroup
-        onActionTriggered: root.actionTriggered(ActivityTypeFilterSubMenu.Send, checked)
-        allChecked: root.allChecked
+        allChecked: d.allChecked
+        checked: typeFilters.includes(Constants.TransactionType.Send)
+        onActionTriggered: root.actionTriggered(Constants.TransactionType.Send)
     }
+
     ActivityTypeCheckBox {
+        id: receiveCheckbox
         title: qsTr("Receive")
-        asset.name: "receive"
+        assetSettings.name: "receive"
         buttonGroup: typeButtonGroup
-        checked: typeFilters.includes(ActivityTypeFilterSubMenu.Receive)
-        onActionTriggered: root.actionTriggered(ActivityTypeFilterSubMenu.Receive, checked)
-        allChecked: root.allChecked
+        allChecked: d.allChecked
+        checked: typeFilters.includes(Constants.TransactionType.Receive)
+        onActionTriggered: root.actionTriggered(Constants.TransactionType.Receive)
     }
+
     ActivityTypeCheckBox {
+        id: buyCheckbox
         title: qsTr("Buy")
-        asset.name: "token"
+        assetSettings.name: "token"
         buttonGroup: typeButtonGroup
-        checked: typeFilters.includes(ActivityTypeFilterSubMenu.Buy)
-        onActionTriggered: root.actionTriggered(ActivityTypeFilterSubMenu.Buy, checked)
-        allChecked: root.allChecked
+        allChecked: d.allChecked
+        checked: typeFilters.includes(Constants.TransactionType.Buy)
+        onActionTriggered: root.actionTriggered(Constants.TransactionType.Buy)
     }
+
     ActivityTypeCheckBox {
+        id: swapCheckbox
         title: qsTr("Swap")
-        asset.name: "swap"
+        assetSettings.name: "swap"
         buttonGroup: typeButtonGroup
-        checked: typeFilters.includes(ActivityTypeFilterSubMenu.Swap)
-        onActionTriggered: root.actionTriggered(ActivityTypeFilterSubMenu.Swap, checked)
-        allChecked: root.allChecked
+        allChecked: d.allChecked
+        checked: typeFilters.includes(Constants.TransactionType.Swap)
+        onActionTriggered: root.actionTriggered(Constants.TransactionType.Swap)
     }
+
     ActivityTypeCheckBox {
```

Comment:
This `ActivityTypeCheckBox` (or sth derived from it) could have a specific property for `Constants.TransactionType.Bridge` and have all the logic done inside it, instead of repeating it below for every instance, wdyt?

Reply:
I think adding a type to the ActivityTypeCheckBox is a great idea, but dont think I can move much more within the component itself, or I'd rather not add complexity to that component further

Reply:
You can still inherit it and add the extra features there

Reply:
yes but the benefit is not too much , i still need to propogate the events further, will only reduce a single line